src/sbt-test/laika/blog-post/project/CustomDirectives.scala

```diff
@@ -24,16 +25,17 @@ object CustomDirectives extends StandardDirectives {
         config.getString("by")
       else ""
 
-    context.map { context =>
-      val posts = context.root.navigatables.collect {
-        case pd: DocumentTree =>
-          pd.documents.map { d =>
-            val path = d.path
-            val refPath = context.parent.path
-            PostsTocElement(titleOrName(d),
-              by(d.config),
-              PathInfo.fromPath(path, refPath.parent),
-              Styles("toc"))
+    cursor.map { cursor =>
+      val posts = cursor.root.children.collect {
+        case pd: TreeCursor =>
+          pd.children.map {
```

Comment:
Sorry, I corrected my comment afterwards. `collect` is safer here as the above might cause match errors.